wolfssl/wolfcrypt/aes.h

```diff
@@ -373,15 +373,23 @@ WOLFSSL_API int wc_AesEcbDecrypt(Aes* aes, byte* out,
 #endif
 /* AES-DIRECT */
 #if defined(WOLFSSL_AES_DIRECT)
-#ifdef WOLFSSL_LINUXKM
- WOLFSSL_API __must_check int wc_AesEncryptDirect(Aes* aes, byte* out, const byte* in);
- WOLFSSL_API __must_check int wc_AesDecryptDirect(Aes* aes, byte* out, const byte* in);
-#else
+#if defined(HAVE_FIPS) && \
+    (!defined(HAVE_FIPS_VERSION) || (HAVE_FIPS_VERSION < 2))
  WOLFSSL_API void wc_AesEncryptDirect(Aes* aes, byte* out, const byte* in);
  WOLFSSL_API void wc_AesDecryptDirect(Aes* aes, byte* out, const byte* in);
-#endif
- WOLFSSL_API int  wc_AesSetKeyDirect(Aes* aes, const byte* key, word32 len,
+ WOLFSSL_API int wc_AesSetKeyDirect(Aes* aes, const byte* key, word32 len,
                                 const byte* iv, int dir);
+#elif defined(BUILDING_WOLFSSL)
+ WOLFSSL_API WARN_UNUSED_RESULT int wc_AesEncryptDirect(Aes* aes, byte* out, const byte* in);
```

Comment:
Are you sure you need the WARN_UNUSED_RESULT here?
I was able to compile without.

Reply:
`WARN_UNUSED_RESULT` is just a safeguard against misuse by callers, current or (here the only concern) future.  Adding it can never transform uncompilable/warned code into compilable/unwarned code -- it can only do the opposite.

Here, I added it to all `static` non-`void` functions in aes.c to lint my work percolating the errors, and guard against future flubs.